lib/extensions/ar_virtual.rb

```diff
@@ -70,13 +70,32 @@ def virtual_delegate(*methods)
       unless (to = options[:to])
         raise ArgumentError, 'Delegation needs an association. Supply an options hash with a :to key as the last argument (e.g. delegate :hello, to: :greeter).'
       end
-      delegate(*methods, options)
+
+      if options[:name] && methods.size > 1
+        raise ArgumentError, 'Option :name is only available when defining a single virtual method'
+      end
+
+      location = caller_locations(1, 1).first
+      file, line = location.path, location.lineno
 
       # put method entry per method name.
       # This better supports reloading of the class and changing the definitions
       methods.each do |method|
         method_prefix = virtual_delegate_name_prefix(options[:prefix], options[:to])
-        method_name = "#{method_prefix}#{method}"
+        method_name = options[:name] || "#{method_prefix}#{method}"
+
+        # delegate(method, options)
```

Comment:
This is a pretty important [comment in `activesupport`](https://github.com/rails/rails/blob/d66e7835/activesupport/lib/active_support/core_ext/module/delegation.rb#L175-L177), do you think it should be here?